### PR TITLE
KOGITO-1714, KOGITO-1717, KOGITO-1713: Enabling subfolder DMN filesearching and fixed item sorting

### PR DESCRIPTION
**Pull request by yesamer**
@danielzhe @dupliaka Can you please test and review it? 

Screenshot from 2020-04-09 12-37-56

This PR solves 3 tickets:
https://issues.redhat.com/browse/KOGITO-1713
https://issues.redhat.com/browse/KOGITO-1714
https://issues.redhat.com/browse/KOGITO-1717

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/dropdown/ScenarioSimulationKogitoRuntimeCreationAssetsDropdownProviderImpl.java [hunk 1]
```diff
@@ -64,6 +66,8 @@ public void getItems(Consumer<List<KieAssetsDropdownItem>> assetListConsumer) {
     }
 
     protected KieAssetsDropdownItem getKieAssetsDropdownItem(final String asset) {
-        return new KieAssetsDropdownItem(asset, "", asset, new HashMap<>());
+        int idx = asset.replaceAll("\\\\", "/").lastIndexOf('/');
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/dropdown/ScenarioSimulationKogitoRuntimeCreationAssetsDropdownProviderImpl.java:69]: What is the reason for this? Can you add a simple test for this class?

Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/dropdown/ScenarioSimulationKogitoRuntimeCreationAssetsDropdownProviderImpl.java:69]: The scope is to extract the filename from a fullPath string.
I'll provide tests.

**Comment by danielzhe** [comment]
How do I enable Scesim inside VSCode? Thank you. @yesamer

**Comment by yesamer** [comment]
> How do I enable Scesim inside VSCode? Thank you. @yesamer
You need this: https://github.com/kiegroup/kogito-tooling/pull/73

**Comment by yesamer** [comment]
> @Test
>     public void remoteCallBackSortedIgnoreCase(){
>         RemoteCallback<List<String>> remoteCallBack = scenarioSimulationKogitoRuntimeCreationAssetsDropdownProviderImplSpy.getRemoteCallback(assetConsumer);
>         List<String> dmnFiles = Arrays.asList("AS", "Ap");
>         remoteCallBack.callback(dmnFiles);
>         verify(assetConsumer, times(1)).accept(dropDownListCaptor.capture());
>         Collections.sort(dmnFiles, String::compareToIgnoreCase);
>         assertEquals(dmnFiles,
>                      dropDownListCaptor.getValue().stream()
>                              .map(KieAssetsDropdownItem::getValue).collect(Collectors.toList()));
>     }

@dupliaka Added, thank you for contributing :)

**Comment by danielzhe** [comment]
Thank you, @yesamer !

**Comment by danielzhe** [comment]
Looks like it's not working on Windows, but I think it's an issue related to content api and not your PR:
scesim

**Comment by yesamer** [comment]
@danielzhe Thank you, for testing in windows and for raise it. @pefernan WDYT about it?

**Comment by ederign** [comment]
@yesamer there is some API changes that we did (traversal method for searches)  that Pere will confirm with you next week.

**Comment by yesamer** [comment]
@ederign @pefernan Ok, thank you!

**Comment by yesamer** [comment]
Hi @pefernan! I opened a new ticket https://issues.redhat.com/browse/KOGITO-1883 to manage this specfic issue, in this way I can proceed to fix this case at least for unix like environments. Thank you!

**Comment by yesamer** [comment]
@danielezonca can you please merge it?

**Comment by yesamer** [comment]
@danielezonca Can you please merge it? THanks!

**Pull request merged**